credentials/src/main/kotlin/web5/sdk/credentials/model/VcDataModel.kt

```diff
@@ -0,0 +1,346 @@
+      return mapper.readValue(json, VcDataModel::class.java)
+    }
+  }
+}
+
+/**
+ * The [CredentialSubject] represents the value of the credentialSubject property as a set of objects containing
+ * properties related to the subject of the verifiable credential.
+ */
+public class CredentialSubject(
+  public val id: URI?,
+  public val additionalClaims: Map<String, Any>
+) {
+  /**
+   * Builder class for creating [CredentialSubject] instances.
+   */
+  public class Builder {
+    private var id: URI? = null
+    private var additionalClaims: Map<String, Any> = emptyMap()
+
+    /**
+     * Sets the ID URI for the credential subject.
+     * @param id The unique identifier URI of the credential subject.
+     * @return Returns this builder to allow for chaining.
+     */
+    public fun id(id: URI?): Builder = apply { this.id = id }
+
+    /**
+     * Sets the additional claims for the credential subject.
+     * Additional claims provide more information about the credential subject.
+     * @param additionalClaims A map of claim names to claim values.
+     * @return Returns this builder to allow for chaining.
+     */
+    public fun additionalClaims(additionalClaims: Map<String, Any>): Builder =
+      apply { this.additionalClaims = additionalClaims }
+
+    /**
+     * Builds and returns the [CredentialSubject] object.
+     * @return The constructed [CredentialSubject] object.
+     * @throws IllegalStateException If the ID URI is not valid.
+     */
+    public fun build(): CredentialSubject {
+      require(id == null || id.toString().isNotBlank()) { "ID URI cannot be blank" }
+
+      return CredentialSubject(id, additionalClaims)
+    }
+  }
+}
+
+
+/**
+ * The [CredentialSchema] Represents the schema defining the structure of a credential.
+ */
+public class CredentialSchema(
+  public val id: String,
+  public val type: String?
+) {
+  /**
+   * Builder class for creating [CredentialSchema] instances.
+   */
+  public class Builder {
+    private var id: String? = null
+    private var type: String? = null
+
+    /**
+     * Sets the ID for the credential schema.
+     * @param id The unique identifier of the credential schema.
+     * @return Returns this builder to allow for chaining.
+     */
+    public fun id(id: String): Builder = apply { this.id = id }
+
+    /**
+     * Sets the type for the credential schema.
+     * @param type The type of the credential schema.
+     * @return Returns this builder to allow for chaining.
+     */
+    public fun type(type: String?): Builder = apply { this.type = type }
+
+    /**
+     * Builds and returns the [CredentialSchema] object.
+     * @return The constructed [CredentialSchema] object.
+     * @throws IllegalStateException If the id is not set.
+     */
+    public fun build(): CredentialSchema {
+      require(!id.isNullOrBlank()) { "ID cannot be blank" }
+      require(type == null || type!!.isNotBlank()) { "Type cannot be blank" }
```

Comment:
<img width="788" alt="image" src="https://github.com/TBD54566975/web5-kt/assets/5314059/ade5a679-1de1-47f4-b27c-a3a4c06ca04d">

hardcode this type

Reply:
can you explain this? i can't tell if this is a todo for you or something i need to understand 😅

Reply:
sorry yes this is a todo for me, need to add a few requires() for credentialSchema for validation